src/LanguageServer.Common/Utilities/DotnetInfo.cs

```diff
@@ -0,0 +1,698 @@
+        /// <remarks>
+        ///     This can happen if the app specified a framework reference to a lower-level framework which is also specified by a higher-level framework which is also used by the app.
+        ///     For example, this would happen if the app referenced Microsoft.AspNet.App version 2.0 and Microsoft.NETCore.App version 3.0.
+        ///     In such case the Microsoft.AspNet.App has .runtimeconfig.json which also references Microsoft.NETCore.App but it only allows versions 2.0 up to 2.9 (via roll forward options).
+        ///     So the version 3.0 requested by the app is incompatible.
+        /// </remarks>
+        FrameworkCompatFailure = unchecked((int)0x8000809c),
+
+        /// <summary>
+        ///     Feature which requires certain version of the hosting layer binaries was used on a version which doesn't support it.
+        /// </summary>
+        /// <remarks>
+        ///     For example if a COM component specified to run on 2.0 Microsoft.NETCore.App - as that contains older version of <c>hostpolicy</c> which doesn't support the necessary features to provide COM services.
+        /// </remarks>
+        HostApiUnsupportedVersion = unchecked((int)0x800080a2),
+        
+        /// <summary>
+        ///     Error returned by <c>hostfxr_initialize_for_runtime_config</c> if the component being initialized requires framework which is not available or incompatible with the frameworks loaded by the runtime already in the process.
+        /// </summary>
+        /// <remarks>
+        ///     For example trying to load a component which requires 3.0 into a process which is already running a 2.0 runtime.
+        /// </remarks>
+        CoreHostIncompatibleConfig = unchecked((int)0x800080a5),
+    }
+
+    /// <summary>
+    ///     Extension methods for <see cref="DotnetInfo"/> and related types.
+    /// </summary>
+    public static class DotnetInfoExtensions
+    {
+        /// <summary>
+        ///     Determine whether a .NET host exit code is well-known (i.e. a defined member of <see cref="DotNetHostExitCode"/>).
+        /// </summary>
+        /// <param name="hostExitCode">
+        ///     The .NET host exit code.
+        /// </param>
+        /// <returns>
+        ///     <c>true</c>, if the exit code represents a well-known host exit code; otherwise, <c>false</c>.
+        /// </returns>
+        public static bool IsWellKnownExitCode(this DotNetHostExitCode hostExitCode) => Enum.IsDefined(hostExitCode);
+
+        /// <summary>
+        ///     Determine whether a .NET host exit code represents success.
+        /// </summary>
+        /// <param name="hostExitCode">
+        ///     The .NET host exit code.
+        /// </param>
+        /// <returns>
+        ///     <c>true</c>, if the exit code represents success; otherwise, <c>false</c>.
+        /// </returns>
+        public static bool IsSuccess(this DotNetHostExitCode hostExitCode) => hostExitCode >= DotNetHostExitCode.Success;
```

Comment:
It is hard to understand the logic here because you've written all exit codes in `0x` notation, making it very unclear, that all error codes are actually negative

Reply:
I originally put them in as negative numbers but switched to hex because almost all instances of these  error results I see reported are in hex format. But given our language server is now returning the enum member name (if defined) you’re probably right and decimal format would be better.